File: ios/Classes/Renderer/BackgroundRenderer.swift
import Foundation

final class BackgroundRenderer: ChartLayerRenderer {
  func render(frame: ChartFrame) {
    frame.context.setFillColor(frame.theme.backgroundColor)
    frame.context.fill(CGRect(x: 0, y: 0, width: frame.width, height: frame.height))
  }
}

File: android/src/main/kotlin/com/tradechart/plugin/renderer/BackgroundRenderer.kt
package com.tradechart.plugin.renderer

import com.tradechart.plugin.engine.ChartFrame

class BackgroundRenderer : ChartLayerRenderer {
    override fun render(frame: ChartFrame) {
        frame.canvas.drawRect(
            0f,
            0f,
            frame.width.toFloat(),
            frame.height.toFloat(),
            frame.theme.backgroundPaint,
        )
    }
}
